osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/threads/RecoverKeyProcess.java

```diff
@@ -139,10 +152,14 @@ private void promoteInvalidKey() {
      *             When there are problems in connecting to or communicating
      *             with the device.
      */
-    private DlmsConnection createConnection() throws IOException, FunctionalException {
-        final byte[] authenticationKey = Hex
-                .decode(this.getSecurityKey(SecurityKeyType.E_METER_AUTHENTICATION).getKey());
-        final byte[] encryptionKey = Hex.decode(this.getSecurityKey(SecurityKeyType.E_METER_ENCRYPTION).getKey());
+    private DlmsConnection createConnectionUsingNewKeys() throws IOException, FunctionalException {
+
+        //TODO: specify that the NEW keys should be retrieved
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1135)